elasticdl/python/worker/worker.py

```diff
@@ -477,11 +528,36 @@ def run(self):
             dataset = dataset.batch(self._minibatch_size).prefetch(1)
             for dataset_batch in dataset:
                 if self._job_type == JobType.TRAINING_WITH_EVALUATION:
-                    self._process_eval_task_if_needed()
+                    if self._process_eval_task_if_needed():
+                        last_minibatch_do_evaluation = True
                 task = self._task_data_service.get_current_task()
+                if task.type == elasticdl_pb2.TRAINING:
+                    if (
+                        last_minibatch_do_evaluation
+                        or last_training_minibatch_failed
+                        or local_update_count >= self._get_model_steps
+                    ):
+                        local_update_count = 0
+                        train_with_local_model = False
+                    else:
+                        train_with_local_model = True
                 err_msg = self._process_minibatch_and_report(
-                    dataset_batch, task.type, task.model_version
+                    dataset_batch,
+                    task.type,
+                    task.model_version,
+                    train_with_local_model,
                 )
+                if task.type == elasticdl_pb2.TRAINING:
+                    last_minibatch_do_evaluation = False
+                    local_update_count += 1
+                    if err_msg:
+                        last_training_minibatch_failed = True
+                    else:
+                        last_training_minibatch_failed = False
+                        if local_update_count < self._get_model_steps:
+                            self._update_local_model()
+                elif task.type == elasticdl_pb2.EVALUATION:
+                    last_minibatch_do_evaluation = True
```

Comment:
This will be evaluation-only job, so no need to set here.

Reply:
Good point. Done.